docker/Dockerfile

```
@@ -0,0 +1,294 @@
+
+ARG ONEAPI_INSTALLER=https://registrationcenter-download.intel.com/akdlm/IRC_NAS/18673/l_BaseKit_p_2022.2.0.262.sh
+ARG ONEAPI_INSTALL_DIR=/opt/intel/oneapi
+
+
+# Some build steps requires a high enough version of cmake.
+# Bump it if necessary.
+
+ARG CMAKE_VERSION=3.23
+ARG CMAKE_VERSION_BUILD=2
+
+
+# Versions of the intel python packages
+
+ARG DPCTL_GIT_BRANCH=0.13.0dev2
+ARG DPCTL_GIT_URL=https://github.com/IntelPython/dpctl.git
+
+ARG DPNP_GIT_BRANCH=master
+ARG DPNP_GIT_URL=https://github.com/IntelPython/dpnp.git
+
+ARG NUMBA_DPEX_GIT_BRANCH=0.18.0
+ARG NUMBA_DPEX_GIT_URL=https://github.com/IntelPython/numba-dpex.git
+
+
+FROM $BASE AS oneapi_base_toolkit_installer
+ARG TMPDIR
+ARG ONEAPI_INSTALL_DIR
+ARG ONEAPI_INSTALLER
+ARG ONEAPI_CACHE_DIR=$TMPDIR/intel/cache
+ARG ONEAPI_DOWNLOAD_DIR=$TMPDIR/intel/download
+ARG ONEAPI_LOG_DIR=$TMPDIR/intel/log
+# TODO: are all those components necessary ?
+# TODO: some of those components that are necessary for build might not be
+# necessary for runtime ?
+ARG ONEAPI_COMPONENTS="intel.oneapi.lin.dpcpp-ct:intel.oneapi.lin.dpcpp_dbg:intel.oneapi.lin.dpcpp-cpp-compiler:intel.oneapi.lin.tbb.devel:intel.oneapi.lin.mkl.devel"
+RUN mkdir -p $ONEAPI_INSTALL_DIR $ONEAPI_CACHE_DIR $ONEAPI_DOWNLOAD_DIR ONEAPI_LOG_DIR \
+    && wget -P $ONEAPI_DOWNLOAD_DIR $ONEAPI_INSTALLER \
+    && chmod +x $ONEAPI_DOWNLOAD_DIR/l_BaseKit_p_2022.2.0.262.sh \
+    && $ONEAPI_DOWNLOAD_DIR/l_BaseKit_p_2022.2.0.262.sh -a -s --eula accept --action install --components $ONEAPI_COMPONENTS --install-dir $ONEAPI_INSTALL_DIR --log-dir $ONEAPI_INSTALL_DIR --download-cache $ONEAPI_CACHE_DIR \
+    && rm -Rf $TMPDIR/*
+
+
+# The two following blocks script a convenient way of installing python
+# It installs a python binary that is separate from the system binary
+# and the version is easily configurable.
+# The build and the environment management is managed with pyenv.
+
+FROM $BASE AS python_base_config
+ENV PYENV_ROOT="/opt/pyenv"
+ENV VIRTUAL_ENV=/opt/venv
+ENV PATH="$VIRTUAL_ENV/bin:$PYENV_ROOT/bin:$PYENV_ROOT/shims:$PATH"
+ENV LD_LIBRARY_PATH="$VIRTUAL_ENV/lib:$LD_LIBRARY_PATH"
+
+FROM python_base_config AS python_builder
+ARG PYTHON_VERSION
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update --quiet; apt-get install -y --no-install-recommends \
+        make \
+        build-essential \
+        libssl-dev \
+        libbz2-dev \
+        libreadline-dev \
+        libsqlite3-dev \
+        wget \
+        curl \
+        llvm \
+        libncurses5-dev \
+        xz-utils \
+        tk-dev \
+        libxml2-dev \
+        libxmlsec1-dev \
+        libffi-dev \
+        liblzma-dev \
+        git \
+        ca-certificates \
+     && rm -rf /var/lib/apt/lists/*
+RUN curl -L \
+    https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer \
```

Should we get this file from a specific tag or commit? e.g:
```suggestion
    https://github.com/pyenv/pyenv-installer/raw/49fba599e872bc761858ea6f700271fb6dcb5a97/bin/pyenv-installer \
```